src/Utils/TypeInfo.php

```diff
@@ -204,8 +205,8 @@ public static function extractTypes($type, ?array $typeMap = null)
             $nestedTypes = array_merge($nestedTypes, $type->getInterfaces());
         }
 
-        if ($type instanceof ObjectType || $type instanceof InterfaceType) {
-            foreach ($type->getFields() as $fieldName => $field) {
+        if ($type instanceof HasFieldsType) {
```

Comment:
This popped out nicely 👍🏾

Reply:
Yeah that is arguably why the interface is the better choice here!